snakemake/executors/__init__.py

```diff
@@ -788,11 +788,11 @@ def format_job(self, pattern, job, **kwargs):
 
             waitfiles_parameter = format(
                 "--wait-for-files-file {wait_for_files_file}",
-                wait_for_files_file=wait_for_files_file,
+                wait_for_files_file=('"' + wait_for_files_file + '"'),
```

Comment:
For quoting a string, you can just use `repr`, which will also automatically take care of escaping quotes that are already occuring within the strings.

Reply:
Added, see below. Thank you.